### PR TITLE
Setting up a developer environment

### PR DESCRIPTION
**Pull request by jasonwilliams**
- This sets up an .editorConfig file, for editor consistency
- Developer environment using docker, so we're using the same versions of Ruby and don't need to install any dependencies manually.
- Set up Jekyll Serve, for easier dev

### CODE REVIEW

**Review by codehag — APPROVED** [review]
this looks quite good to me. Re: framework -- we haven't settled on one really yet, would you be opposed to updating this once we do?

File: .editorConfig [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[**.js]
+indent_style = space
+indent_size = 2
+
+[**.css]
+indent_style = tab
```
[the remark this answers is not in the document]
Comment (jasonwilliams) [reply .editorConfig]: Yep, this is fine, i can change that

**Comment by jasonwilliams** [comment]
> this looks quite good to me. Re: framework -- we haven't settled on one really yet, would you be opposed to updating this once we do?

Sure, this is easily changeable, so long as there's an easy way to get set up without installing all the dependencies

**Comment by jasonwilliams** [comment]
@codehag updated to spaces

File: .editorConfig (hunk 1 above)
[the remark this answers is not in the document]
Comment (ljharb) [reply .editorConfig]: :-p tabs > spaces tho

**Pull request merged**